features/cloud.py

```diff
@@ -462,3 +476,108 @@ def _create_instance(
         print("--- Launching Azure image {}({})".format(image_name, series))
         inst = self.api.launch(image_id=image_name, user_data=user_data)
         return inst
+
+
+class LXD(Cloud):
+    name = "lxd"
+
+    @property
+    def api(self) -> pycloudlib.cloud.BaseCloud:
+        """Return the api used to interact with the cloud provider."""
+        if self._api is None:
+            self._api = pycloudlib.LXD(tag=self.tag)
+
+        return self._api
+
+    def manage_ssh_key(self, private_key_path: "Optional[str]" = None) -> None:
+        """Create and manage ssh key pairs to be used in the cloud provider.
+
+        :param private_key_path:
+            Location of the private key path to use. If None, the location
+            will be a default location.
+        """
+        pub_key_path = "lxd-pubkey"
```

Comment:
As you mentioned in passing, the integration tests currently take 4 times as long running via ssh to the lxd instance instead of lxc exec. Should this method be a noop (pass) so that lxd testing doesn't use ssh by default during typical per-PR integration tests?